client/src/main/java/com/battleships/gamewindow/board/Ship.java

```diff
@@ -17,11 +17,41 @@ public Ship(Set<BoardField> masts) {
         return masts;
     }
 
-    public boolean isMastHit(Coordinate coordinate) {
-        return masts.stream().anyMatch(m -> m.getCoordinate().equals(coordinate));
+    boolean isMastHit(Coordinate coordinate) {
+        return getMastForCoordinate(coordinate)
+                .orElseGet(() -> new BoardField(Coordinate.fromIntCoords(0, 0), FieldState.SEA))
+                .isHit();
     }
 
-    public String toString(){
+    private Optional<BoardField> getMastForCoordinate(Coordinate coordinate) {
+        return masts.stream().filter(boardField -> boardField.getCoordinate().equals(coordinate))
+                .findFirst();
+    }
+
+    boolean isSunk() {
+        boolean isSunk = masts.stream().allMatch(BoardField::isHit);
+        if (isSunk)
+            sunkShip();
```

Comment:
The method name isSunk() does not says that it sunks fleet

Reply:
fixed